### PR TITLE
GH54: Enhance addin installation

### LINKED ISSUE #54: Enhance addin installation
It takes lots of time to shell out to nuget and try to install packages which doesn't exist.

Example, writing `#addin Cake.Figlet` in VSCode one char at a time will result in trying to install packages `C`, `Ca`, `Cak` etc...

### PR DESCRIPTION
**Pull request by mholo65**
- Fixes #54

### CODE REVIEW

File: src/Cake.Bakery/Program.cs [hunk 1]
```diff
@@ -14,13 +14,15 @@
...
+using Cake.Core.Scripting;
 using Cake.Core.Text;
 using Cake.NuGet;
 using Cake.Scripting;
 using Cake.Scripting.Abstractions;
-using Cake.Scripting.CodeGen;
 using Cake.Scripting.Transport.Tcp.Server;
 using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using IScriptAliasFinder = Cake.Scripting.CodeGen.IScriptAliasFinder;
```
Comment (patriksvensson) [inline src/Cake.Bakery/Program.cs]: Why a using alias here?

File: src/Cake.Bakery/Scripting/CachingScriptProcessor.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+                    filePaths.AddRange(result);
+                }
+            }
+            return filePaths;
+        }
+
+        public void InstallTools(IReadOnlyCollection<PackageReference> tools, DirectoryPath installPath)
+        {
+            // We don't need to actually install tools with bakery.
+            return;
```
Comment (patriksvensson) [inline src/Cake.Bakery/Scripting/CachingScriptProcessor.cs]: Return statement not necessary here.

File: src/Cake.Bakery/Scripting/PackageReferenceComparer.cs [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+            if (x == null && y == null)
+            {
+                return true;
+            }
+            if (x == null || y == null)
+            {
+                return false;
+            }
+
+            return x.OriginalString.Equals(y.OriginalString, StringComparison.OrdinalIgnoreCase);
```
Comment (patriksvensson) [inline src/Cake.Bakery/Scripting/PackageReferenceComparer.cs]: `x.OriginalString` could theoretically be null here. Change to `string.Equals(x,y,comparison)`.

**Comment by mholo65** [comment]
@patriksvensson fixed!

**Comment by mholo65** [comment]
@patriksvensson also added so that addins are only installed upon save. Haven't tested this yet, so please don't merge :wink:

**Comment by patriksvensson** [comment]
Are you telling me to merge? 😊

**Comment by mholo65** [comment]
No, the opposite :wink:

**Comment by patriksvensson** [comment]
@mholo65 Ok, I will merge ASAP and delete the branch!!!!

**Comment by mholo65** [comment]
@patriksvensson feel free to merge whenever you like.

**Comment by patriksvensson** [comment]
@mholo65 Thank you for this (minor) contribution to this - to you - completely new project. We hope to see more contributions from you in the future. If you need any explanation about how any of this stuff work, feel free to reach out. ❤️ 👍

**Pull request merged**